proto/interchain_security/ccv/v1/ccv.proto

```diff
@@ -56,17 +56,22 @@ message SlashPacketData {
 // unbonding operations.
 message MaturedUnbondingOps { repeated uint64 ids = 1; }
 
-// ConsumerPacketData contains a consumer packet data and a type tag
+// ConsumerPacketData contains a consumer packet data, type tag, and index for storage.
 message ConsumerPacketData {
   ConsumerPacketDataType type = 1;
 
   oneof data {
     SlashPacketData slashPacketData = 2;
     VSCMaturedPacketData vscMaturedPacketData = 3;
   }
+  uint64 idx = 4;
 }
 
+
+// [Depreciated] favor using []ConsumerPacketData directly, which can be stored more efficiently.
+//
 // ConsumerPacketDataList is a list of consumer packet data packets.
+// It is only used for genesis to ensure backwards compatibility with older versions of ICS.
```

Comment:
Why is this needed for backwards compability? IIRC, this list is exported in `ExportGenesis`. If a chain runs `ExportGenesis` with ICS v2.0.0 for example, then it should run `InitGenesis` with the same version. I don't think we need to keep backwards compability here.

Reply:
Your understanding is all correct 👍 however the issue becomes a little more complex when you consider that consumers query genesis state from the provider. 

If the provider is running v2 for example, and a consumer is running v4 with the PR merged, there would be manual changes we'd have to make to the consumer genesis JSON. Similar to Stride and adding the `PreCCV` field. Are we okay with manual changes being required? 

It's a tradeoff between keeping around some technical debt vs requiring some hackyness if the provider is on an old version of ICS

Reply:
That's a good point. I keep forgetting that the consumer CCV genesis state is part of the "wire".  Then I guess it's good to use `ConsumerPacketDataList` to export the genesis state. Please do not mark the struct as deprecated though. Just explain in the docstring that it's used only for exporting / importing the state.

Reply:
SGTM -> https://github.com/cosmos/interchain-security/commit/a42229a465fd6f8f3f4fc14dcb4a5285ba1c4b84